src/sentry/integrations/vsts/issues.py

```diff
@@ -156,3 +157,28 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
                     'exception': error,
                 }
             )
+
+    def should_unresolve(self, data):
+        return self.is_done(data['old_category']) and not self.is_done(data['new_category'])
+
+    def should_resolve(self, data):
+        return not self.is_done(data['old_category']) and self.is_done(data['new_category'])
+
+    def get_state_categories(self, old_state, new_state, project):
+        client = self.get_client()
+        all_states = client.get_work_item_states(
+            self.instance, project)['value']
+        new_category = None
+        old_category = None
+        for state in all_states:
+            if state['name'] == new_state:
+                new_category = state['category']
+            if state['name'] == old_state:
+                old_category = state['category']
+
+        # what should I do if this doesn't work?
+        assert old_category is not None and new_category is not None
+        return old_category, new_category
```

Comment:
wondering if you could simplify this by doing something like:

```python
def get_done_states(self, project):
    if not self._done_states:
        client = self.get_client()
        all_states = client.get_work_item_states(self.instance, project)['value']
        self._done_states = [
            state['name'] for state in all_states if state['category'] in self.done_categories
        ]
    return self._done_states
```

then you could just use that in `should_resolve`/`should_unresolve`?  what do you think? just trying to think of ways to simplify/contain the logic since right now it's spread across this and the webhook